http-clients/url-connection-client/src/main/java/software/amazon/awssdk/http/urlconnection/UrlConnectionHttpClient.java

```diff
@@ -302,7 +302,10 @@ public HttpExecuteResponse call() throws IOException {
             int responseCode = getResponseCodeSafely(connection);
             boolean isErrorResponse = HttpStatusFamily.of(responseCode).isOneOf(CLIENT_ERROR, SERVER_ERROR);
             Optional<InputStream> responseContent = isErrorResponse ? tryGetErrorStream() : tryGetInputStream();
-            AbortableInputStream responseBody = responseContent.map(AbortableInputStream::create).orElse(null);
+
+            AbortableInputStream responseBody = responseHasNoContent()
```

Comment:
Should we do this in tryGetErrorStream and tryGetInputStream (or their downstream methods)? It seems weird that we have responseContent being present even when there's technically no content.

Reply:
- Agree, thats because of https://github.com/JetBrains/jdk8u_jdk/blob/master/src/share/classes/sun/net/www/protocol/http/HttpURLConnection.java#L1858
- Thus it make sense in adding in tryGetInputStream
- However in case of `tryGetErrorStream` we get null Stream in case of errors for all type of method like 'GET' , 'HEAD' etc , so I think it not required for `tryGetErrorStream`